### ISSUE TITLE
openssl fails to build without makedepend

### ISSUE THREAD

**Issue by OJFord** [Linuxbrew/homebrew-core#1684]
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with specific formulae and not Homebrew/brew?

I'm not sure what [this](https://github.com/Linuxbrew/homebrew-core/blob/master/Formula/openssl.rb#L34-L45) is all about:
```ruby
  if OS.mac?
    depends_on "makedepend" => :build
  else
    depends_on "zlib"
    depends_on :perl => ["5.0", :build]
  end

  # Build without makedepend.
  patch do
    url "https://github.com/openssl/openssl/pull/1524.patch"
    sha256 "853a15755d3bc2b561d64c4f5d58a7ff4885e5cdbacbf429ce93b477e0d3d68e"
  end unless OS.mac?
```

Why those conditionals are beneficial or thought to be needed that is, because `openssl` won't build for me unless I manually `brew install makedepend` first; then it's fine.

**Comment by sjackman** [comment]
Please report `brew gist-logs openssl`

**Comment by fibric** [comment]
I had an issue today `brew doctor` complained about foreign *-config files but I didn't know them and I decided to uninstall everything and install `yarn`, `git`, `nvm`, `caddy`, and `rethinkdb`. But the installation of `git` failed with an error regarding `openssl`. I cleaned up everything again and just tried installing `openssl` alone.

My gist-log https://gist.github.com/anonymous/8cea459730180234ceb7d29f46680c50
